src/scmdata/units.py

```diff
@@ -394,50 +367,83 @@ def result_backward(ur, strt):
                     ).items()
                 ][0]
 
-                unit_reg_unit = getattr(
+                base_unit_ureg = getattr(
                     self, base_unit.replace("[", "").replace("]", "")
                 )
-                tc.add_transformation(
-                    base_unit, "[carbon]", _get_transform_func(unit_reg_unit, conv_val)
-                )
-                tc.add_transformation(
-                    "[carbon]",
-                    base_unit,
-                    _get_transform_func(unit_reg_unit, conv_val, forward=False),
-                )
-                tc.add_transformation(
-                    "[mass] * {} / [time]".format(base_unit),
-                    "[mass] * [carbon] / [time]",
-                    _get_transform_func(unit_reg_unit, conv_val),
-                )
-                tc.add_transformation(
-                    "[mass] * [carbon] / [time]",
-                    "[mass] * {} / [time]".format(base_unit),
-                    _get_transform_func(unit_reg_unit, conv_val, forward=False),
-                )
-                tc.add_transformation(
-                    "[mass] * {}".format(base_unit),
-                    "[mass] * [carbon]",
-                    _get_transform_func(unit_reg_unit, conv_val),
-                )
-                tc.add_transformation(
-                    "[mass] * [carbon]",
-                    "[mass] * {}".format(base_unit),
-                    _get_transform_func(unit_reg_unit, conv_val, forward=False),
-                )
-                tc.add_transformation(
-                    "{} / [time]".format(base_unit),
-                    "[carbon] / [time]",
-                    _get_transform_func(unit_reg_unit, conv_val),
-                )
-                tc.add_transformation(
-                    "[carbon] / [time]",
-                    "{} / [time]".format(base_unit),
-                    _get_transform_func(unit_reg_unit, conv_val, forward=False),
+
+                tc = self._add_transformations_to_context(
+                    tc, base_unit, base_unit_ureg, "[carbon]", other_unit_ureg, conv_val
                 )
 
             self.add_context(tc)
 
+    @staticmethod
+    def _add_transformations_to_context(
+        context, base_unit, base_unit_ureg, other_unit, other_unit_ureg, conv_val
+    ):
+        """
+        Add all the transformations between units (mass x unit per time, mass x unit
+        etc. to a context for the two given units)
+        """
+
+        def _get_transform_func(forward=True):
+            if forward:
+
+                def result_forward(ur, strt):
+                    return strt * other_unit_ureg / base_unit_ureg * conv_val
+
+                return result_forward
+
+            def result_backward(ur, strt):
+                return strt * base_unit_ureg / other_unit_ureg / conv_val
```

Comment:
Can you add brackets to make the order of operation explicit. I had to look up python's order of operations to double check...